src/sentry/backup/findings.py

```diff
@@ -196,9 +198,14 @@ class FindingJSONEncoder(json.JSONEncoder):
 
     def default(self, obj):
         if isinstance(obj, Finding):
-            d = deepcopy(obj.__dict__)
-            kind = d.get("kind")
+            keys = set(Finding.__annotations__.keys())
+            kind = getattr(obj, "kind", None)
+            d = {k: obj.__dict__[k] for k in keys if k in obj.__dict__}
```

Comment:
Digging through an objects `__dict__` from outside the class is generally a pattern we should avoid as you could access internal state that is added in the future. Could we have a method that exposes the necessary information instead? We have a few objects that implement a `to_dict` method.

Reply:
Done.